dot/rpc/modules/author_integration_test.go

```diff
@@ -179,148 +231,552 @@ func TestAuthorModule_SubmitExtrinsic_invalid_input(t *testing.T) {
 	expected := &transaction.ValidTransaction{
-		Extrinsic: types.NewExtrinsic(testExt),
-		Validity:  val,
+		Extrinsic: expectedExtrinsic,
+		Validity: &transaction.Validity{
+			Priority:  39325240425794630,
+			Requires:  nil,
+			Provides:  [][]byte{{212, 53, 147, 199, 21, 253, 211, 28, 97, 20, 26, 189, 4, 169, 159, 214, 130, 44, 133, 88, 133, 76, 205, 227, 154, 86, 132, 231, 165, 109, 162, 125, 0, 0, 0, 0}}, // nolint:lll
+			Longevity: 18446744073709551614,
+			Propagate: true,
+		},
 	}
 
-	_, err := txQueue.Push(expected)
-	require.Nil(t, err)
-
-	// this should cause error since transaction is already in txQueue
-	err = auth.SubmitExtrinsic(nil, &ext, res)
-	require.EqualError(t, err, transaction.ErrTransactionExists.Error())
+	integrationTestController.stateSrv.Transaction.AddToPool(expected)
 
+	// should not cause error, since a transaction
+	err := auth.SubmitExtrinsic(nil, &ext, res)
+	require.NoError(t, err)
 }
 
-func TestAuthorModule_InsertKey_Valid(t *testing.T) {
-	seed := "0xb7e9185065667390d2ad952a5324e8c365c9bf503dcf97c67a5ce861afe97309"
-	kp, err := sr25519.NewKeypairFromSeed(common.MustHexToBytes(seed))
-	require.NoError(t, err)
+func TestAuthorModule_InsertKey_Integration(t *testing.T) {
+	tmppath := t.TempDir()
+	integrationTestController := setupStateAndRuntime(t, tmppath, useInstanceFromGenesis)
+	auth := newAuthorModule(t, integrationTestController)
 
-	auth := setupAuthModule(t, nil)
-	req := &KeyInsertRequest{"babe", seed, kp.Public().Hex()}
-	res := &KeyInsertResponse{}
-	err = auth.InsertKey(nil, req, res)
-	require.Nil(t, err)
-	require.Len(t, *res, 0) // zero len result on success
-}
+	const seed = "0xb7e9185065667390d2ad952a5324e8c365c9bf503dcf97c67a5ce861afe97309"
 
-func TestAuthorModule_InsertKey_Valid_Gran_Keytype(t *testing.T) {
-	seed := "0xb7e9185065667390d2ad952a5324e8c365c9bf503dcf97c67a5ce861afe97309"
-	kp, err := ed25519.NewKeypairFromSeed(common.MustHexToBytes(seed))
+	babeKp, err := sr25519.NewKeypairFromSeed(common.MustHexToBytes(seed))
 	require.NoError(t, err)
 
-	auth := setupAuthModule(t, nil)
-	req := &KeyInsertRequest{"gran", seed, kp.Public().Hex()}
-	res := &KeyInsertResponse{}
-	err = auth.InsertKey(nil, req, res)
-	require.Nil(t, err)
+	grandKp, err := ed25519.NewKeypairFromSeed(common.MustHexToBytes(seed))
+	require.NoError(t, err)
 
-	require.Len(t, *res, 0) // zero len result on success
-}
+	testcases := map[string]struct {
+		ksType, seed string
+		kp           interface{}
+		waitErr      error
+	}{
+		"insert a valid babe key type": {
+			ksType: "babe",
+			seed:   seed,
+			kp:     babeKp,
+		},
+
+		"insert a valid gran key type": {
+			ksType: "gran",
+			seed:   seed,
+			kp:     grandKp,
+		},
+
+		"invalid babe key type": {
+			ksType:  "babe",
+			seed:    seed,
+			kp:      "0x0000000000000000000000000000000000000000000000000000000000000000",
+			waitErr: errors.New("generated public key does not equal provide public key"),
```

Comment:
Declare an error variable for this  in `dot/rpc/modules/author.go` and use that in tests (here and in author_test.go)